src/DataSource/ArrayDataSource.php

```diff
@@ -93,6 +93,9 @@ public function filterOne(array $condition): IDataSource
 	{
 		foreach ($this->data as $item) {
 			foreach ($condition as $key => $value) {
+				if ($key === 'id' && is_string($value)) {
+					$value = intval($value);
+				}
 				if ($item[$key] === $value) {
```

Comment:
```suggestion
				if ($item[$key] == $value) {
```

What about to change to `==` to ignore types?

Reply:
Yes i thought of that too, its better idea a should work better. Guess there is no chance of getting like `value = null` and having $item[0] for match. 
Another thing that went over my head is regex check for only numbers in value then convert to to int.

But ignoring types is easiest stuff to do. Thank you

Reply:
Are you going to change the pull request?

Reply:
Pull request updated.